### DISCUSSION TITLE
Allow Lovelace cards to add automations, scripts and helpers

### DISCUSSION THREAD

**Discussion by chaptergy** [home-assistant/frontend#7350] [Other feature requests]
I think Lovelace cards could cover a much wider range if they had the ability to define automations, scripts and helpers by themselves. For example if a card is for a much more abstract feature than just editing an entity (for example my lightalarm-card) this would be very helpful, so the user doesn't have to manually create all things required for such a card to work

**Comment by iantrich** [comment]
There's nothing really stopping a card from all ready doing this as far as I know. You just need to look through the codebase to find the proper API/WS calls to list/create/delete them

**Reply by iantrich** [reply]
e.g. https://github.com/home-assistant/frontend/blob/dev/src/panels/config/automation/ha-automation-editor.ts#L722

**Reply by KTibow** [reply]
In that case is this a HA dev documentation issue?

**Comment by iantrich** [comment]
@KTibow the method is documented but not all endpoints
https://developers.home-assistant.io/docs/frontend/data#hasscallwsmessage

**Comment by thomasloven** [comment]
Another method to do this would be to create a lightalarm *integration* and have that create the necessary helpers in the backend. The integration can also serve the frontend card code through its own endpoint on the HA webserver.

I believe Dwain (of [Dwains theme](https://github.com/dwainscheeren/dwains-lovelace-dashboard) fame) is working on something like this...
